libnavigation-core/src/main/java/com/mapbox/navigation/core/adasis/ADASISv2Message.kt

```diff
@@ -0,0 +1,37 @@
+package com.mapbox.navigation.core.adasis
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+
+/**
+ * Describes a class holding methods for ADASISv2 messages.
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+class ADASISv2Message internal constructor(
+    private val nativeMessage: com.mapbox.navigator.ADASISv2Message
```

Comment:
I'm not sure how OK it is to store a reference to a native object here. We give ADASISv2Message to the user, they may keep it when they are not required anymore and there'll be problems with native memory (like that JNI reference overflow). 
I think this is not critical for the snapshot as long as we don't plan on changing the API much if we change the implementation in the future.

Reply:
We don't have much choice here. One options it to do like this with storing reference to a native object and alternative is to call `toFlatBuffer()`, `toJson()`, etc. in advance and store their result. These two options come with their own disadvantages. I'll talk to NN team and ask if this can be optimised.